### PR TITLE
PWN-3992 - settings_network_values feature toggle 

### PR DESCRIPTION
**Pull request by gslevinkov**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-3992

## Description of Work

- add usage of settings_network_values feature-toggle

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/common/feature_toggles/di/FeatureTogglesModule.kt [hunk 1]
```diff
@@ -14,7 +14,7 @@ import org.p2p.wallet.common.feature_toggles.toggles.remote.SslPinningFeatureTog
 
 object FeatureTogglesModule : InjectionModule {
     override fun create() = module {
-        singleOf(::AppFirebaseRemoteConfig)
+        single(createdAtStart = true) { AppFirebaseRemoteConfig() }
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/common/feature_toggles/di/FeatureTogglesModule.kt:17]: now RemoteConfig is initialised when the app starts without any explicit method call in `App` or other place

File: app/src/main/java/org/p2p/wallet/common/feature_toggles/toggles/remote/SettingsNetworkListFeatureToggle.kt [hunk 2]
```diff
@@ -21,7 +21,12 @@ class SettingsNetworkListFeatureToggle(
     override val featureKey: String = "settings_network_values"
     override val featureDescription: String = "List of available networks to choose from"
 
-    override val defaultValue: List<SettingsNetworkValue> = emptyList()
+    override val defaultValue: List<SettingsNetworkValue> = listOf(
+        SettingsNetworkValue(url = "https://p2p.rpcpool.com", networkName = "mainnet-beta"),
+        SettingsNetworkValue(url = "https://solana-api.projectserum.com", networkName = "mainnet-beta"),
+        SettingsNetworkValue(url = "https://api.mainnet-beta.solana.com", networkName = "mainnet-beta"),
+        SettingsNetworkValue(url = "https://api.devnet.solana.com/", networkName = "devnet")
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/common/feature_toggles/toggles/remote/SettingsNetworkListFeatureToggle.kt:28]:
```suggestion
```

File: app/src/main/java/org/p2p/wallet/debugdrawer/ConfigurationModule.kt [hunk 3]
```diff
@@ -8,17 +8,17 @@ import org.koin.core.component.KoinComponent
 import org.koin.core.component.inject
 import org.p2p.wallet.common.InAppFeatureFlags
 import org.p2p.wallet.databinding.ViewDebugDrawerConfigureEnvironmentBinding
-import org.p2p.wallet.infrastructure.network.environment.EnvironmentManager
+import org.p2p.wallet.infrastructure.network.environment.NetworkEnvironmentManager
 
 class ConfigurationModule : DebugModuleAdapter(), KoinComponent {
 
-    private val environmentManager: EnvironmentManager by inject()
+    private val environmentManager: NetworkEnvironmentManager by inject()
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/debugdrawer/ConfigurationModule.kt:15]: renamed due to this entity being responsible for *Network* Environment exactly, but not some abstract Environment as a whole

File: app/src/main/java/org/p2p/wallet/infrastructure/network/environment/NetworkEnvironmentManager.kt [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+    var availableNetworks: List<NetworkEnvironment> = listOf(
+        NetworkEnvironment.MAINNET,
+        NetworkEnvironment.RPC_POOL,
+        NetworkEnvironment.SOLANA
+    )
+        private set
+    private var listeners = mutableMapOf<String, EnvironmentManagerListener>()
+
+    fun loadAvailableEnvironments(networks: List<NetworkEnvironment>) {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/infrastructure/network/environment/NetworkEnvironmentManager.kt:27]: load networks fetched from remote config

File: app/src/main/java/org/p2p/wallet/moonpay/api/MoonpayUrlCreator.kt [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+import org.p2p.wallet.utils.Constants.USD_READABLE_SYMBOL
+
+private const val QUERY_API_KEY = "apiKey"
+private const val QUERY_CURRENCY_CODE = "currencyCode"
+private const val QUERY_BASE_CURRENCY_CODE = "baseCurrencyCode"
+private const val QUERY_BASE_CURRENCY_AMOUNT = "baseCurrencyAmount"
+private const val QUERY_LOCK_AMOUNT = "lockAmount"
+private const val QUERY_WALLET_ADDRESS = "walletAddress"
+
+class MoonpayUrlCreator(private val moonpayWalletDomain: String, private val moonpayApiKey: String) {
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/moonpay/api/MoonpayUrlCreator.kt:13]:
```suggestion
MoonpayUrlBuilder
```

File: app/src/main/java/org/p2p/wallet/root/RootPresenter.kt [hunk 6]
```diff
@@ -21,6 +27,15 @@ class RootPresenter(
         }
     }
 
+    override fun loadAvailableNetworkEnvironments() {
+        environmentManager.loadAvailableEnvironments(networkListFeatureToggle.value.toDomain())
+    }
+
+    private fun List<SettingsNetworkValue>.toDomain(): List<NetworkEnvironment> {
+        val allNetworks = NetworkEnvironment.values()
+        return mapNotNull { toggleNetwork -> allNetworks.find { it.endpoint == toggleNetwork.url } }
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/root/RootPresenter.kt:36]: weird mapping. 
Why you just check url instead of passing all params?

**Review by 1sakov — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/p2p/wallet/moonpay/api/MoonpayUrlCreator.kt (hunk 5 above)
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/moonpay/api/MoonpayUrlCreator.kt:13]: you want an object class is ok?

File: app/src/main/java/org/p2p/wallet/root/RootPresenter.kt (hunk 6 above)
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/root/RootPresenter.kt:36]: oh, correct, i will replace it with `filter`

**Review by kabramovp2p — APPROVED** [review]

**Pull request merged**